### DISCUSSION TITLE
有没有发现使用 ref 实现 useForm, 太依赖于状态变更?

### DISCUSSION THREAD

**Discussion by vaynevayne** [ant-design/ant-design#43650] [Ideas]
antd 的 useForm 是使用 ref 来实现的, 缺点:  在 render 里取不到ref 的值, 编辑弹窗时,setFieldsValue(row), 会排斥 id 属性, 因为使用 formItem组件 来注册依赖关系, 我一般不专门做性能优化, 所以这个特性倒成了 bug 的滋生环境 

```jsx
return (
// 这是无效的, 必须二次封装 input, 在内部使用 Form.useWatch
<input  value={form.getFiledValue('name')}/>
)
```

有时候使用起来太笨重,
加上 FormItem 绑定死了 value 属性, 导致想放到 <span>里完全做不到, 
自带的几种 layout布局是放到Form api 上, 但是响应式布局很难调整, 我觉得 formItem 太多余了,  直接 form.getInputProps, 就很简单的完成绑定

以上是一些个人的思考,如果我来设计我会这么做

**Comment by zombieJ** [comment]
Form 渲染是惰性的，在 v3 的时候就是每次变更都会全部渲染一遍，导致性能很差。在 v4 以后只会控制变化部分。降级成全刷新就是 useWatch 根节点就好了。

**Reply by vaynevayne** [reply]
前提:  `setFieldsvalue` 的值, 应该跟 `getFieldsValue` 获取的一致, 
开启性能优化, 他们就不一致了, 其依赖于你组件有没有使用到这个 field
我觉得性能优化时,不应该修改这个前提

**Reply by zombieJ** [reply]
`setFieldsValue` 和 `getFieldsValue` 是一致的。在 `setFieldsValue` 后，调用 `getFieldsValue` 获得的就是新的值哈~

**Reply by vaynevayne** [reply]
会排斥 id 属性

**Reply by zombieJ** [reply]
这个排斥 id 属性举个例子看看？

**Reply by vaynevayne** [reply]
之前有过一个问题啊，说是需要加一个hidden的input

**Reply by linxianxi** [reply]
<img width="1281" alt="image" src="https://github.com/ant-design/ant-design/assets/47104575/cc25da72-f0cb-4cb7-8412-b9cdce437932">
认真看文档，加 true

**Reply by vaynevayne** [reply]
这接口设计的
